runtime/src/ceramic/RoundedQuad.hx

```diff
@@ -0,0 +1,65 @@
+package ceramic;
+
+import ceramic.Shape;
+
+/**
+ * An extension of Shape that creates a nicely rounded rectangle
+ */
+class RoundedQuad extends Shape {
+	/**
+	 * Amount of corner segments
+	 * One to ten is going to be the sanest quantity
+	 */
+	public var segments:Int = 10;
+	/**
+	 * Defines the radius of each corner
+	 */
+	public var radius:CornerRadius = {tl: 0, tr: 0, br: 0, bl: 0};
+
+	override public function new() {
+		super();
+		autoComputeSize = false;
+	}
+
+	override function computeContent() {
+		points = [];
+
+		// Define the relative coordinates for the radius
+		var sine = [for (angle in 0...segments + 1) Math.sin(Math.PI / 2 * angle / segments)];
+		var cosine = [for (angle in 0...segments + 1) Math.cos(Math.PI / 2 * angle / segments)];
+
+		// TOP LEFT
+		for (pointPairIndex in 0...segments) {
+			points.push(radius.tl * (1 - cosine[pointPairIndex]));
+			points.push(radius.tl * (1 - sine[pointPairIndex]));
+		}
+
+		// TOP RIGHT
+		for (pointPairIndex in 0...segments) {
+			points.push(width + radius.tr * (cosine[segments - pointPairIndex] - 1));
+			points.push(radius.tr * (1 - sine[segments - pointPairIndex]));
+		}
+
+		// BOTTOM RIGHT
+		for (pointPairIndex in 0...segments) {
+			points.push(width + radius.br * (cosine[pointPairIndex] - 1));
+			points.push(height + radius.br * (sine[pointPairIndex] - 1));
+		}
+
+		// BOTTOM LEFT
+		for (pointPairIndex in 0...segments) {
+			points.push(radius.bl * (1 - cosine[segments - pointPairIndex]));
+			points.push(height + radius.bl * (sine[segments - pointPairIndex] - 1));
+		}
+		
+		super.computeContent();
+
+	}
+}
+
+typedef CornerRadius = {
```

Comment:
Using a typedef for CornerRadius here adds unnecessary overhead, especially on C++ target which will treat that anonymous structure as a map with dynamic string key access.

The most "Ceramic-way" of doing this would be to have 4 fields directly on the `RoundedQuad` class like `radiusTopLeft`, `radiusTopRight`, `radiusBottomRight` and `radiusBottomLeft` and a shortcut method `radius()` that set all four properties at once.

You can see a similar example of what I'm talking about in the `View` class with the `padding*` fields: https://github.com/ceramic-engine/ceramic/blob/23efeaa805c186fab8f9f1bfc50def6aa4d80530/plugins/ui/runtime/src/ceramic/View.hx#L61-L121